src/Impostor.Server/Events/Game/Player/PlayerMovementEvent.cs

```diff
@@ -2,23 +2,46 @@
 using Impostor.Api.Games;
 using Impostor.Api.Net;
 using Impostor.Api.Net.Inner.Objects;
+using Microsoft.Extensions.ObjectPool;
 
 namespace Impostor.Server.Events.Player
 {
-    // TODO: Finish and use event, needs to be pooled
-    public class PlayerMovementEvent : IPlayerEvent
+    public class PlayerMovementEvent : IPlayerMovementEvent
     {
-        public PlayerMovementEvent(IGame game, IClientPlayer clientPlayer, IInnerPlayerControl playerControl)
+#pragma warning disable 8766
```

Comment:
?

Reply:
A small problem with object pools, IPlayerEvent expects Game, ClientPlayer to be always not null, and from a plugin you should never get null from these, but they have to be null while in the pool

Reply:
Remove the pragma lines

Reply:
pragma lines > yellow crying msbuild